client/packages/programs/src/JsonForms/common/components/CategorizationTabLayout.tsx

```diff
@@ -140,7 +140,10 @@ const containsErrors = (layout: Layout, errorPaths: string[]): Set<string> => {
         ''
       );
       for (const errorPath of errorPaths) {
-        if (scopePath == errorPath) {
+        // Match on the parent as well, as some components have internal "child"
+        // elements that set the error rather than the component itself
+        const errorPathParent = errorPath.split('/').slice(0, -1).join('/');
+        if (scopePath == errorPath || scopePath === errorPathParent) {
```

Comment:
I simplified the BloodPressure component by removing a messy `useEffect` update, but the consequence is that *only* the child elements (two data leaves below the component level) show in error state, so this change just loosens the error matching here in order to display the "number of errors" badge on the Category tile.

<img width="167" height="161" alt="Image" src="https://github.com/user-attachments/assets/4608ace0-4ef5-4958-a8e2-8e40bbbace7f" />